src/file_organizer/parallel/processor.py

```
@@ -400,36 +541,83 @@ def _handle_timeouts(
+                    # ``non_retryable=True`` and discarding its actual return
+                    # value.
                     pending.remove(future)
-                    completed_path = future_paths.pop(future)
-                    future_started.pop(future, None)
+                    del future_paths[future]
+                    del future_started[future]
+                    future_queued_at.pop(future, None)
                     try:
-                        completed_result = finalize_result(future.result())
+                        race_result = future.result()
                     except Exception as exc:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)